### ISSUE TITLE
Verify and fix compatibility with different PowerShell Core versions

### ISSUE THREAD

**Issue by lupino3** [microsoft/PowerShell-FeatureFlags#36]
As discovered in https://github.com/microsoft/PowerShell-FeatureFlags/pull/35, it looks like loading NJsonSchema will fail if it conflicts with the version shipped with PowerShell.

We should verify what versions of PowerShell Core we are currently compatible with, and ideally work around any limitations.

**Comment by lupino3** [comment]
The current code works with PowerShell >= 7.2.0-rc1 (tested 7.2.0-rc1, 7.2.0, 7.2.2, 7.2.5).
Previous versions fail with the following error

```
Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'.
```

**Comment by lupino3** [comment]
Note that the error for PowerShell 6 before 6.1.0 is a Pester error, not an error from this library.

I tried manually importing Newtonsoft.Json, and got the very same error:

```
PS /app> Write-Host $PSVersionTable.PSVersion
6.0.4
PS /app> Add-Type -Path ./External/newtonsoft.json/13.0.1/lib/netstandard1.0/Newtonsoft.Json.dll
Add-Type : Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'.
At line:1 char:1
+ Add-Type -Path ./External/newtonsoft.json/13.0.1/lib/netstandard1.0/N ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [Add-Type], FileLoadException
+ FullyQualifiedErrorId : System.IO.FileLoadException,Microsoft.PowerShell.Commands.AddTypeCommand
```

I don't think there is an easy way to be compatible with PowerShell 6.0.x, so I'll close this issue and open a new one to track future work to improve this compatibility.

**Comment by lupino3** [comment]
Reopening as I actually found and fixed the issue 😮‍💨 

Now Pester is complaining about running under that version of Powershell.

**Comment by lupino3** [comment]
Note that #39 fixed compatibility with PowerShell Core >= 6.1.0.
